Sources/GroundState/py_variational_montecarlo.cc

```diff
@@ -0,0 +1,309 @@
+          R"EOF(netket.machine.Machine: The machine used to express the wavefunction.)EOF")
+      .def("add_observable", &VariationalMonteCarlo::AddObservable,
+           py::keep_alive<1, 2>(), py::arg("ob"), py::arg("ob_name"), R"EOF(
+           Add an observable quantity, that will be calculated at each
+           iteration.
+
+           Args:
+               ob: The operator form of the observable.
+               ob_name: The name of the observable.
+
+           )EOF")
+      .def("run", &VariationalMonteCarlo::Run, py::arg("output_prefix"),
+           py::arg("n_iter") = nonstd::nullopt, py::arg("step_size") = 1,
+           py::arg("save_params_every") = 50, R"EOF(
+           Optimize the Vmc wavefunction.
+
+           Args:
+               output_prefix: The output file name, without extension.
+               n_iter: The maximum number of iterations.
+               step_size: Number of iterations performed at a time. Default is 1.
+               save_params_every: Frequency to dump wavefunction parameters. The
+                   default is 50.
+
+           Examples:
+               Running a simple Vmc calculation.
+
+
+               ```python
+               >>> import netket as nk
+               >>> SEED = 3141592
+               >>> g = nk.graph.Hypercube(length=8, n_dim=1)
+               >>> hi = nk.hilbert.Spin(s=0.5, graph=g)
+               >>> ma = nk.machine.RbmSpin(hilbert=hi, alpha=1)
+               >>> ma.init_random_parameters(seed=SEED, sigma=0.01)
+               >>> ha = nk.operator.Ising(hi, h=1.0)
+               >>> sa = nk.sampler.MetropolisLocal(machine=ma)
+               >>> sa.seed(SEED)
+               >>> op = nk.optimizer.Sgd(learning_rate=0.1)
+               >>> vmc = nk.variational.Vmc(hamiltonian=ha, sampler=sa,
+               ... optimizer=op, n_samples=500)
+               >>> vmc.run(output_prefix='test', n_iter=1)
+
+
+               ```
+
+           )EOF")
+      .def("reset", &VariationalMonteCarlo::Reset)
+      .def("advance", &VariationalMonteCarlo::Advance, py::arg("steps") = 1,
+           R"EOF(
+           Perform one or several iteration steps of the VMC calculation. In each step,
+           energy and gradient will be estimated via VMC and subsequently, the variational
+           parameters will be updated according to the configured method.
+
+           Args:
+               steps: Number of VMC steps to perform.
+
+           )EOF")
+      .def(
+          "get_observable_stats",
+          [](VariationalMonteCarlo &self) {
+            self.ComputeObservables();
+            return self.GetObservableStats();
+          },
+          R"EOF(
+        Calculate and return the value of the operators stored as observables.
+
+        )EOF")
+      .def_property_readonly("vmc_data", &VariationalMonteCarlo::GetVmcData)
+      .def_property_readonly("sr", &VariationalMonteCarlo::GetSR);
+
+  m_vmc.def(
+      "compute_samples",
```

Comment:
compute_samples should be in `netket.sampler.compute_samples` as the corresponding C++ function!

Reply:
done